src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector256.cs

```diff
@@ -2351,6 +2353,45 @@ public static Vector256<sbyte> Shuffle(Vector256<sbyte> vector, Vector256<sbyte>
             return result;
         }
 
+        /// <summary>Creates a new vector by selecting values from an input vector using a set of indices.
+        /// Behavior is platform-dependent for out-of-range indices.</summary>
+        /// <param name="vector">The input vector from which values are selected.</param>
+        /// <param name="indices">The per-element indices used to select a value from <paramref name="vector" />.</param>
+        /// <returns>A new vector containing the values from <paramref name="vector" /> selected by the given <paramref name="indices" />.</returns>
+        /// <remarks>Unlike Shuffle, this method delegates to the underlying hardware intrinsic without ensuring that <paramref name="indices"/> are normalized to [0, 31].
+        /// On hardware with <see cref="Avx2"/> support, indices are treated as modulo 32, and if the high bit is set, the result will be set to 0 for that element.
+        /// On hardware with <see cref="AdvSimd.Arm64"/> support, this method behaves the same as Shuffle.</remarks>
+        /// On hardware with <see cref="PackedSimd"/> support, indices are treated as modulo 32.
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        [CompExactlyDependsOn(typeof(Avx2))]
+        [CompExactlyDependsOn(typeof(AdvSimd))]
+        [CompExactlyDependsOn(typeof(AdvSimd.Arm64))]
+        [CompExactlyDependsOn(typeof(PackedSimd))]
+        public static Vector256<byte> ShuffleUnsafe(Vector256<byte> vector, Vector256<byte> indices)
+        {
+            if (Avx2.IsSupported)
+            {
+                return Avx2.Shuffle(vector, indices);
+            }
+
+            if (AdvSimd.Arm64.IsSupported)
+            {
+                (Vector128<byte>, Vector128<byte>) table = (vector.GetLower(), vector.GetUpper());
+                return Vector256.Create(
+                    AdvSimd.Arm64.VectorTableLookup(table, indices.GetLower()),
+                    AdvSimd.Arm64.VectorTableLookup(table, indices.GetUpper()));
+            }
+
+            if (PackedSimd.IsSupported)
+            {
+                return Vector256.Create(
+                    PackedSimd.Shuffle(vector.GetLower(), vector.GetUpper(), indices.GetLower()),
+                    PackedSimd.Shuffle(vector.GetLower(), vector.GetUpper(), indices.GetUpper()));
+            }
```

Comment:
We shouldn't need this additional handling for `V256`/`V512`. It isn't expected that users are calling these, let alone expecting good performance, when `IsHardwareAccelerated` reports `false`.

It's just causing the JIT to spend more time doing dead code elimination, managing the inliner budget, etc.